### PR TITLE
Handle child processes

### LINKED ISSUE #429: [BUG] LocalTestCluster fails during create and destroy
**Describe the bug**
The LocalTestCluster is not working with recent opensearch tarballs and fails to launch the OpenSearch process successfully. There are a couple of sub-issues here

1. The OpenSearch process is not setup properly.
2. The PID of the OpenSearch process seems to be reported incorrectly. It is the pid of shell subprocess that starts OpenSearch process. 
2. Missing logs to debug the opensearch process setup failures. The opensearch-tar-install.sh script logs are not available in stdout for debugging failure.

**To Reproduce**
1. Stub the download() method and provide a pre-downloaded and uncompressed tarball. Then use LocalTestCluster `create()` method to spin the cluster. It is the `create()` method where problem is seen.

**Expected behavior**
OpenSearch cluster should setup correctly and querying on port 9200 should respond. The opensearch logs should be reported on stdout for debugging.

**Plugins**
Full bundle - ran test with security.

**Screenshots**

**Host/Environment (please complete the following information):**
 - OS: Linux
 - Version = 1.1.0

### PR DESCRIPTION
**Pull request by gaiksaya**
Signed-off-by: Sayali Gaikawad <gaiksaya@amazon.com>

### Description
The PID of the actual opensearch process was different from parent PID in ubuntu. 
This change checks for the child processes and kills them before terminating the actual PID.
In case of AL2 and MACOS, both actual process and PID are same. 
 
### Issues Resolved
#429 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by saratvemulapalli — DISMISSED** [review]
Looks good to me. Thanks @gaiksaya !

File: bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import subprocess
 import time
 
+import psutil  # type: ignore
```
Comment (saratvemulapalli) [inline bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py:12]: Curious why you are trying to ignore?

Comment (dblock) [reply bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py:12]: these modules come without types and pypy is complaining

**Review by dblock — DISMISSED** [review]

File: bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py [hunk 2]
```diff
@@ -101,6 +102,14 @@ def wait_for_service(self):
         raise ClusterCreationException("Cluster is not green after 10 attempts")
 
     def terminate_process(self):
+        parent = psutil.Process(self.process.pid)
+        logging.info("Checking for child processes")
```
Comment (dblock) [inline bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py:106]: Should probably be debug.

File: bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py [hunk 3]
```diff
@@ -101,6 +102,14 @@ def wait_for_service(self):
...
+        child_processes = parent.children(recursive=True)
+        for child in child_processes:
+            logging.info(f"Found child process with pid {child.pid}")
+            if child.pid != self.process.pid:
+                logging.info(f"Sending SIGKILL to {child.pid} ")
```
Comment (dblock) [inline bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py:111]: Same, debug. These don't add much value in the output.

File: bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py (hunk 2 above)
Comment (gaiksaya) [reply bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py]: Changed to debug

File: bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py (hunk 3 above)
Comment (gaiksaya) [reply bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py]: Changed to debug

File: bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py (hunk 1 above)
Comment (saratvemulapalli) [reply bundle-workflow/src/test_workflow/integ_test/local_test_cluster.py:12]: Interesting..

**Review by saratvemulapalli — APPROVED** [review]

**Review by peternied — APPROVED** [review]

**Pull request merged**